pkg/client/intermediate/intermediate.go

```diff
@@ -0,0 +1,139 @@
+package intermediate
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"regexp"
+	"sort"
+
+	"github.com/bitnami-labs/charts-syncer/pkg/client"
+
+	"github.com/juju/errors"
+
+	"github.com/bitnami-labs/charts-syncer/internal/utils"
+	"github.com/bitnami-labs/charts-syncer/pkg/client/types"
+	"helm.sh/helm/v3/pkg/chart"
+)
+
+var (
+	versionRe = regexp.MustCompile("(.*)-(\\d+\\.\\d+\\.\\d+)\\.bundle.tar")
+)
+
+// Repo allows to operate a chart repository.
+type Repo struct {
+	dir     string
+	entries map[string][]string
```

Comment:
Why is each map entry mapping to an array of strings? what is the key and the value of this map really?

Reply:
Oh I see, several versions of the same chart...

Maybe we should use a type to document that:
~~~go
type chartVersions []string
...
type Repo struct {
	dir     string
	entries map[string]chartVersions
...
~~~